src/modules/tas_studio/mod.rs

```diff
@@ -385,6 +388,37 @@ fn convert_hltas_from_bxt_tas_new(marker: MainThreadMarker, mut path: String) {
     )
 }
 
+static BXT_TAS_STUDIO_NOREFRESH_UNTIL_STOP_FRAME: CVar = CVar::new(
+    b"_bxt_tas_studio_norefresh_until_stop_frame\0",
+    b"1\0",
+    "\
+When enabled, TAS Studio playback will stop norefresh when reaching some values set in \
+bxt_tas_norefresh_until_last_frames counting from stop frame marker instead of last frame.",
+);
+
+fn norefresh_until_stop_frame_frame_idx(marker: MainThreadMarker, editor: &Editor) -> usize {
+    if !BXT_TAS_STUDIO_NOREFRESH_UNTIL_STOP_FRAME.as_bool(marker) {
+        return 0;
+    };
+
+    let norefresh_last_frames = unsafe { bxt::BXT_TAS_NOREFRESH_UNTIL_LAST_FRAMES.get(marker)() };
+    (editor.stop_frame() as usize).saturating_sub(norefresh_last_frames as usize)
+}
+
+fn set_effective_norefresh_until_stop_frame(marker: MainThreadMarker, editor: &Editor) {
+    let value = if BXT_TAS_STUDIO_NOREFRESH_UNTIL_STOP_FRAME.as_bool(marker) {
+        // Very sad
+        match editor.branch().frames.len().checked_sub(1) {
```

Comment:
Isn't there always the initial frame there? Don't remember

Reply:
I am not sure what you mean. If you mean the frames length, then yea I have to subtract 1. That is what also happens to the status hud last frame pr

Reply:
What I mean is, can this checked sub ever fail?

Reply:
From what I remember when I did this, this check sub is here just because unsigned. If it fails, it gets to 0. In BunnymodXT PR, 0 for this will just load the entire script. 

Now I think about it again, I don't really remember why I really did it beside the integer overflow. Probably was to make it extra safe. I don't think it will ever happen from my recent usage sessions.

Reply:
It is possible that I was scared out of index because I was working on other PR with bulk_idx function while also working on this so it carries over, maybe I guess there shouldn't be the check sub.

Reply:
FWIW bxt-rs enables integer overflow checks in release. If it ever breaks you will see because it will panic. Just IIRC frames always has at least one entry so checked should be unnecessary.